cmd/cri-containerd/options/options.go

```diff
@@ -99,13 +107,18 @@ func (c *CRIContainerdOptions) AddFlags(fs *pflag.FlagSet) {
 		"/var/run/cri-containerd.sock", "Path to the socket which cri-containerd serves on.")
 	fs.StringVar(&c.RootDir, "root-dir",
 		"/var/lib/cri-containerd", "Root directory path for cri-containerd managed files (metadata checkpoint etc).")
-	fs.StringVar(&c.ContainerdRootDir, "containerd-root-dir",
-		"/var/lib/containerd", "Root directory path where containerd stores persistent data. "+
-			"This should be the same with containerd `root`.")
-	fs.StringVar(&c.ContainerdEndpoint, "containerd-endpoint",
+	fs.StringVar(&c.ContainerdConfig.RootDir, "containerd-root-dir",
+		"/var/lib/containerd", "Root directory path where containerd stores persistent data.")
+	fs.StringVar(&c.ContainerdConfig.Endpoint, "containerd-endpoint",
 		"/run/containerd/containerd.sock", "Path to the containerd endpoint.")
-	fs.StringVar(&c.ContainerdSnapshotter, "containerd-snapshotter",
+	fs.StringVar(&c.ContainerdConfig.Snapshotter, "containerd-snapshotter",
 		containerd.DefaultSnapshotter, "Snapshotter used by containerd.")
+	fs.StringVar(&c.ContainerdConfig.Runtime, "containerd-runtime",
+		"io.containerd.runtime.v1.linux", "runtime used by containerd.")
+	fs.StringVar(&c.ContainerdConfig.RuntimeEngine, "containerd-runtime-engine",
+		"runc", "runtime engine used by containerd.")
+	fs.StringVar(&c.ContainerdConfig.RuntimeRoot, "containerd-runtime-root",
```

Comment:
This seems to be https://github.com/containerd/containerd/blob/fe9d6a46d1e9f4cea358d21bd93bee2ddf9b422d/linux/shim/service.go#L32.

I prefer we set empty default which indicates let containerd use its own default for both RuntimeEngine, and RuntimeRoot.

Reply:
agree on the comment part if null.. but we should talk about the Engine and Root being empty by default..https://github.com/containerd/containerd/blob/master/linux/runtime.go#L449-L460

Reply:
s/the/The.